code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.java

```diff
@@ -1,35 +1,29 @@
 // Part of Cosmos by OpenGenus Foundation
 class LongestCommonSubsequence {
-  /* Returns length of LCS for X[0..m-1], Y[0..n-1] */
-  int lcs( char[] X, char[] Y, int m, int n ) {
-    int L[][] = new int[m+1][n+1];
- 
-    //L[i][j] contains length of LCS of X[0..i-1] and Y[0..j-1]
-    for (int i=0; i<=m; i++) {
-      for (int j=0; j<=n; j++) {
-        if (i == 0 || j == 0) {
-            L[i][j] = 0;
-        } else if (X[i-1] == Y[j-1]) {	// if there is a match
-            L[i][j] = L[i-1][j-1] + 1;	// increate lcs value by 1
-        } else {				// else
-            L[i][j] = Math.max(L[i-1][j], L[i][j-1]);	// choose max value till now
+    /* Returns length of longestCommonSubsequence for X[0..m-1], Y[0..n-1] */
+  public int longestCommonSubsequence(String text1, String text2) {
+        int dp[][] = new int[text1.length() + 1][text2.length() + 1];
+        for (int i = 0; i < text1.length(); i++) {
+            for (int j = 0; j < text2.length(); j++) {
+                if (text1.charAt(i) == text2.charAt(j))
+                    dp[i + 1][j + 1] = dp[i][j] + 1;
+                else
+                    dp[i + 1][j + 1] = Math.max(dp[i][j + 1], dp[i + 1][j]);
+            }
         }
-      }
+        return dp[text1.length()][text2.length()];
     }
-  return L[m][n];
-  }
- 
+
   public static void main(String[] args) {
-    LongestCommonSubsequence lcs = new LongestCommonSubsequence();
-    String s1 = "AGGAGTCTAGCTAB";
-    String s2 = "AGXGTTXAYBATCGAT";
- 
-    char[] X=s1.toCharArray();
-    char[] Y=s2.toCharArray();
-    int m = X.length;
-    int n = Y.length;
- 
-    System.out.println("Length of LCS is" + " " + lcs.lcs( X, Y, m, n ) );
-  }
+        LongestCommonSubsequence lcs = new LongestCommonSubsequence();
+        String s1 = "AGGAGTCTAGCTAB";
+        String s2 = "AGXGTTXAYBATCGAT";
+
+        char []X = s1.toCharArray();
+        char []Y = s2.toCharArray();
+        int m = X.length;
+        int n = Y.length;
+
+        System.out.println("Length of LCS is" + " " + lcs.longestCommonSubsequence(X, Y, m, n));
```

Comment:
Function definition is different. Make sure your code compiles and works correctly.

Reply:
@AdiChat  thanks for your review.
I fix the error.
Now please check the code.